controller/internal/routes/query_options.go

```diff
@@ -31,46 +31,22 @@ const (
 	OffsetDefault = 0
 )
 
-type QueryOptions struct {
+// Represents external query options, which limits some query options (i.e. return all vs max limit). PublicQueryOptions
+// can be used internally as long as the public option limitations are fitting.
+type PublicQueryOptions struct {
 	Predicate string
 	Sort      string
 	Paging    *Paging
 }
 
-func (qo *QueryOptions) String() string {
+func (qo *PublicQueryOptions) String() string {
 	if qo == nil {
 		return "nil"
 	}
 	return fmt.Sprintf("[QueryOption Predicate: '%v', Sort: '%v', Paging: '%v']", qo.Predicate, qo.Sort, qo.Paging)
 }
 
-func (qo *QueryOptions) ValidateAndCorrect() {
```

Comment:
This was long longer called. Most of the logic is handled in the query processing logic.